File: 2021/day-02.rs
use std::{
    fs::File,
    io::{BufRead, BufReader},
};

pub fn main() {
    let file = File::open("src\\day-02-input.txt").unwrap();
    let lines = BufReader::new(file).lines();
    let mut horizontal: i32 = 0;
    let mut depth: i32 = 0;

    let parsed: Vec<_> = lines.map(|l| l.unwrap()).collect();

    for text in &parsed {
        let amount: i32 = text.chars().rev().next().unwrap().to_digit(10).unwrap() as i32;

        if text.contains("forward") {
            horizontal += amount;
        } else if text.contains("down") {
            depth += amount;
        } else if text.contains("up") {
            depth -= amount;
        }
    }

    println!("Part one = {}", horizontal * depth);

    horizontal = 0;
    depth = 0;
    let mut aim = 0;
    for text in &parsed {
        let amount: i32 = text.chars().rev().next().unwrap().to_digit(10).unwrap() as i32;

        if text.contains("forward") {
            horizontal += amount;
            depth += aim * amount;
        } else if text.contains("down") {
            aim += amount;
        } else if text.contains("up") {
            aim -= amount;
        }
    }

    println!("Part two = {}", horizontal * depth);
}

File: 2022/day-02.py
results_p1 = {
    'A': {
        "X": 4,
        "Y": 8,
        "Z": 3
    },
    "B": {
        "X": 1,
        "Y": 5,
        "Z": 9
    },
    "C": {
        "X": 7,
        "Y": 2,
        "Z": 6
    }
}

results_p2 = {
    'A': {
        "X": 3,
        "Y": 4,
        "Z": 8
    },
    "B": {
        "X": 1,
        "Y": 5,
        "Z": 9
    },
    "C": {
        "X": 2,
        "Y": 6,
        "Z": 7
    }
}


def solve():
    input = [x.strip().split(' ') for x in open("day-02-input.txt").readlines()]
    print("Part one = " + str(sum(map(lambda i: results_p1[i[0]][i[1]], input))))
    print("Part two = " + str(sum(map(lambda i: results_p2[i[0]][i[1]], input))))


solve()
